src/nunit.analyzers.tests/ConstraintsUsage/StringConstraintUsageCodeFixTests.cs

```diff
@@ -0,0 +1,39 @@
+using Gu.Roslyn.Asserts;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Analyzers.Constants;
+using NUnit.Analyzers.ConstraintUsage;
+using NUnit.Framework;
+
+namespace NUnit.Analyzers.Tests.ConstraintsUsage
+{
+    public class StringConstraintUsageCodeFixTests
+    {
+        private static readonly DiagnosticAnalyzer analyzer = new StringConstraintUsageAnalyzer();
+        private static readonly CodeFixProvider fix = new StringConstraintUsageCodeFix();
+
+        [TestCase(nameof(string.Contains), AnalyzerIdentifiers.StringContainsConstraintUsage, "Does.Contain")]
+        [TestCase(nameof(string.StartsWith), AnalyzerIdentifiers.StringStartsWithConstraintUsage, "Does.StartWith")]
+        [TestCase(nameof(string.EndsWith), AnalyzerIdentifiers.StringEndsWithConstraintUsage, "Does.EndWith")]
+        public void AnalyzeStringBooleanMethodPositiveAssert(string method, string analyzerId, string suggestedConstraint)
+        {
+            var code = TestUtility.WrapInTestMethod($@"Assert.True(""abc"".{method}(""ab""));");
+
+            var fixedCode = TestUtility.WrapInTestMethod($@"Assert.That(""abc"", {suggestedConstraint}(""ab""));");
+
+            AnalyzerAssert.CodeFix(analyzer, fix, ExpectedDiagnostic.Create(analyzerId), code, fixedCode);
+        }
+
```

Comment:
Perhaps we should also have tests that cover the codefix when the code before the fix is using `Assert.That(...)` instead of `Assert.True(...)`, e.g. `Assert.That("...".StartsWith("..."))`.